keyboards/nifty_numpad/nifty_numpad.c

```diff
@@ -0,0 +1,247 @@
+/* Copyright 2023 Acliad
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef RGB_IDLE_ENABLE
+typedef struct {
+    uint8_t mode;
+    uint8_t speed;
+    HSV     hsv;
+} rgb_matrix_state_t;
+
+typedef struct {
+    rgb_matrix_state_t active_rgb_matrix;
+    rgb_matrix_state_t idle_rgb_matrix;
+} kb_config_t;
+
+static uint32_t idle_timer = 0; 
+bool rgb_idle_mode = false;
+bool rgb_idle_edit_mode = false;
+static rgb_matrix_state_t active_rgb_matrix;
+static rgb_matrix_state_t idle_rgb_matrix;
+static kb_config_t config;
+
+// Returns true if current RGB matrix mode is idle
+bool rgb_matrix_idle_mode(void){
+    return rgb_idle_mode;
+}
+
+// Stores the current mode, HSV, and speed of the RGB matrix into state
+void rgb_matrix_state_save(rgb_matrix_state_t* state){
+    state->mode = rgb_matrix_get_mode();
+    state->hsv = rgb_matrix_get_hsv();
+    state->speed = rgb_matrix_get_speed();
+}
+
+// Restores the mode, HSV, and speed of the RGB matrix from previous state
+void rgb_matrix_state_restore(rgb_matrix_state_t* state){
+    HSV hsv = state->hsv;
+    rgb_matrix_mode_noeeprom(state->mode);
+    rgb_matrix_sethsv_noeeprom(hsv.h, hsv.s, hsv.v);
+    rgb_matrix_set_speed_noeeprom(state->speed);
+}
+#endif
+
+void matrix_scan_kb(void) {
+
+    #ifdef RGB_IDLE_ENABLE
+    // Check if enough time has passed since last keypress to go into idle mode
+    if ( timer_elapsed32(idle_timer) > RGB_IDLE_TIMEOUT_MS && !rgb_idle_mode ) {
+        rgb_matrix_state_save(&active_rgb_matrix);
+        rgb_idle_mode = true;
+        idle_timer = timer_read32();
+
+        rgb_matrix_state_restore(&idle_rgb_matrix);
+    }
+    #endif
+
+    matrix_scan_user();
```

Comment:
```suggestion
```

Reply:
Done as part of change to housekeeping_task_kb()